File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/AttendAllDto.java
package uk.gov.justice.digital.hmpps.whereabouts.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.gov.justice.digital.hmpps.whereabouts.model.TimePeriod;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
@ApiModel(description = "Attend all parameters")
public class AttendAllDto {
    @ApiModelProperty(required = true, value = "Set of active booking ids", example = "1")
    @NotNull
    private Set<Long> bookingIds;

    @ApiModelProperty(required = true, value = "Id of event", example = "2")
    @NotNull
    private Long eventId;

    @ApiModelProperty(required = true, value = "Id of the location the event is taking place", example = "4")
    @NotNull
    private Long eventLocationId;

    @ApiModelProperty(required = true, value = "Time period for the event", example = "AM")
    @NotNull
    private TimePeriod period;

    @ApiModelProperty(required = true, value = "Id of prison the event is taking place", example = "LEI")
    @NotNull
    private String prisonId;

    @JsonFormat(pattern="yyyy-MM-dd")
    @ApiModelProperty(required = true, value = "Date the event is scheduled", example = "2019-10-01")
    @NotNull
    private LocalDate eventDate;

}

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/attendance/AttendAllDto.kt
package uk.gov.justice.digital.hmpps.whereabouts.dto.attendance

import com.fasterxml.jackson.annotation.JsonFormat
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import jakarta.validation.constraints.NotNull
import uk.gov.justice.digital.hmpps.whereabouts.dto.BookingActivity
import uk.gov.justice.digital.hmpps.whereabouts.model.TimePeriod
import java.time.LocalDate

@ApiModel(description = "Attend all parameters")
data class AttendAllDto(
  @ApiModelProperty(required = true, value = "Set of active booking and activity ids", position = 1)
  val bookingActivities: Set<BookingActivity>,

  @ApiModelProperty(required = true, value = "Id of the location the event is taking place", example = "4", position = 2)
  @field:NotNull
  val eventLocationId: Long? = null,

  @ApiModelProperty(required = true, value = "Time period for the event", example = "AM", position = 3)
  @field:NotNull
  val period: TimePeriod? = null,

  @ApiModelProperty(required = true, value = "Id of prison the event is taking place", example = "LEI", position = 4)
  @field:NotNull
  val prisonId: String? = null,

  @JsonFormat(pattern = "yyyy-MM-dd")
  @ApiModelProperty(required = true, value = "Date the event is scheduled", example = "2019-10-01", position = 5)
  @field:NotNull
  val eventDate: LocalDate? = null
)
